pkg/unittest/assertion.go

```diff
@@ -34,14 +35,38 @@ func (a *Assertion) Assert(
 	renderError error,
 	result *results.AssertionResult,
 	failfast bool,
+	didPostRender bool,
```

Comment:
You'll notice this field propagated through the rest of the stack.  We need to indicate whether the post-renderer did something, because if it did there's a chance we need to handle the "one merged file named `manifest.yaml`" case.